cylc/flow/prerequisite.py

```diff
@@ -223,6 +224,45 @@ def dump(self):
         # (Else trigger wiped out by pre-initial simplification.)
         return res
 
+    def api_dump(self, workflow_id):
+        """Return list of populated Protobuf data objects."""
+        if self.conditional_expression:
+            temp = self.get_raw_conditional_expression()
+            temp = temp.replace('|', ' | ')
+            temp = temp.replace('&', ' & ')
+        else:
+            for s_msg in self.satisfied:
+                temp = self.MESSAGE_TEMPLATE % s_msg
+        conds = []
+        num_length = int(math.ceil(float(len(self.satisfied)) / float(10)))
+        for ind, message_tuple in enumerate(sorted(self.satisfied)):
+            name, point = message_tuple[0:2]
+            t_id = f"{workflow_id}/{point}/{name}"
+            char = 'c%.{0}d'.format(num_length) % ind
+            c_msg = self.MESSAGE_TEMPLATE % message_tuple
+            c_val = self.satisfied[message_tuple]
+            c_bool = bool(c_val)
+            if c_val is False:
```

Comment:
We have `c_bool` = bool(c_val)`. You have cast the value of `c_val` to `boolean`, so I am assuming that value may be something else like a `string` or `list`, etc.

But in this `if` condition you are still using `c_val is False`. Is that correct? I suspect it won't ever be true, if it is not already a boolean, as - freshly tested - `"False" is False` results in `False`.

Reply:
Fixed.